File: Leetcode/20. Valid Parentheses.py
class Solution:
    def isValid(self, ss: str) -> bool:
        arr = []
        for i in ss:
            if i in "([{":
                arr.append(i)
            elif len(arr)==0:
                return False
            elif i == "}":
                if arr[-1] == "{":
                    arr = arr[:-1]
                else:
                    arr.append(i)
            elif i == "]":
                if arr[-1] == "[":
                    arr = arr[:-1]
                else:
                    arr.append(i)
            elif i == ")":
                if arr[-1] == "(":
                    arr = arr[:-1]
                else:
                    arr.append(i)                    
                    
        return True if len(arr)==0 else False

File: Leetcode/1-999/20. Valid Parentheses.java
import java.util.Stack;

class Solution {
    public boolean isValid(String str) {
        Stack<Character> stack = new Stack<>();
        boolean status = true;
        for(char i : str.toCharArray()){
            if((i == '{') || (i == '[') || (i == '(')) stack.push(i);
            else if(stack.isEmpty()){
                    status = false;
                    break;
                }
            else if(i == '}'){
                if(stack.peek() == '{') stack.pop();
                else{
                    status = false;
                    break;
                }
            }
            
            else if(i == ']'){
                if(stack.peek() == '[') stack.pop();
                else{
                    status = false;
                    break;
                }
            }
            
            else if(i == ')'){
                if(stack.peek() == '(') stack.pop();
                else{
                    status = false;
                    break;
                }
            }
            
        }
        
        return status&&stack.isEmpty();
    }
}
